FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/include/mbedtls_freertos.h

```diff
@@ -0,0 +1,193 @@
+ */
+typedef struct NetworkCredentials
+{
+    /**
+     * @brief Set this to a non-NULL value to use ALPN.
+     *
+     * This string must be NULL-terminated.
+     *
+     * See [this link]
+     * (https://aws.amazon.com/blogs/iot/mqtt-with-tls-client-authentication-on-port-443-why-it-is-useful-and-how-it-works/)
+     * for more information.
+     */
+    const char * pAlpnProtos;
+
+    /**
+     * @brief Disable server name indication (SNI) for a TLS session.
+     */
+    BaseType_t disableSni;
+
+    const char * pRootCa;     /**< @brief String representing a trusted server root certificate. */
+    size_t rootCaSize;        /**< @brief Size associated with #IotNetworkCredentials.pRootCa. */
+    const char * pClientCert; /**< @brief String representing the client certificate. */
+    size_t clientCertSize;    /**< @brief Size associated with #IotNetworkCredentials.pClientCert. */
+    const char * pPrivateKey; /**< @brief String representing the client certificate's private key. */
+    size_t privateKeySize;    /**< @brief Size associated with #IotNetworkCredentials.pPrivateKey. */
+    const char * pUserName;   /**< @brief String representing the username for MQTT. */
+    size_t userNameSize;      /**< @brief Size associated with #IotNetworkCredentials.pUserName. */
+    const char * pPassword;   /**< @brief String representing the password for MQTT. */
+    size_t passwordSize;      /**< @brief Size associated with #IotNetworkCredentials.pPassword. */
+} NetworkCredentials_t;
+
+/**
+ * @brief mbedTLS Connect / Disconnect return status.
+ */
+typedef enum MbedtlsStatus
+{
+    MBEDTLS_SUCCESS = 0,         /**< Function successfully completed. */
+    MBEDTLS_INVALID_PARAMETER,   /**< At least one parameter was invalid. */
+    MBEDTLS_INSUFFICIENT_MEMORY, /**< Insufficient memory required to establish connection. */
+    MBEDTLS_INVALID_CREDENTIALS, /**< Provided credentials were invalid. */
+    MBEDTLS_HANDSHAKE_FAILED,    /**< Performing TLS handshake with server failed. */
+    MBEDTLS_API_ERROR,           /**< A call to a system API resulted in an internal error. */
+    MBEDTLS_CONNECT_FAILURE      /**< Initial connection to the server failed. */
+} MbedtlsStatus_t;
+
+/**
+ * @brief Create a TCP connection with FreeRTOS sockets.
+ *
+ * @param[out] pNetworkContext Pointer to a network context to contain the
+ * initialized socket handle.
+ * @param[in] pHostName The hostname of the remote endpoint.
+ * @param[in] port The destination port.
+ * @param[in] pNetworkCredentials Credentials for the TLS connection.
+ * @param[in] receiveTimeoutMs Receive socket timeout.
+ * @param[in] sendTimeoutMs Send socket timeout.
+ *
+ * @return #MBEDTLS_SUCCESS, #MBEDTLS_INSUFFICIENT_MEMORY, #MBEDTLS_INVALID_CREDENTIALS,
+ * #MBEDTLS_HANDSHAKE_FAILED, #MBEDTLS_API_ERROR, or #MBEDTLS_CONNECT_FAILURE.
+ */
+MbedtlsStatus_t Mbedtls_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
```

Comment:
As the sockets connect/disconnect has been separated out to avoid use of the `NetworkContext_t` type, should these connect/disconnect functions be separated out as well (or at least avoid using the the type) ?

Reply:
There is an action item to consider that approach. For this PR, @abhidixi11 and I had a discussion and decided to keep using the network context.